extensions/content-manager/admin/src/containers/ListView/index.js

```diff
@@ -347,39 +367,19 @@ function ListView({
     return ids;
   }
 
-  const createRelationStructure = () => {
-    const copiedData = data.map(el => ({ ...el }));
-    copiedData.forEach(el => {
-      let parents = copiedData.filter(x => {
-        if(x[relation]) {
-          return x[relation].id === el.id
-        } else {
-          return false
-        }
-      })
-      if(parents.length) {
-        parents.forEach( parent => {
-          if(parent.id !== el.id) {
-            parent.child = el;
-            el.isChild = true;
-          }
-        })
-      }
-    })
-    const topLevelData = copiedData.filter(el => !el.isChild)
-    return topLevelData;
-  }
-
   const findInitialIds = () => data.map(dataEl => dataEl.id)
-
   const initialIds = findInitialIds();
-  const relationData = createRelationStructure(data);
-  const idsFromNestedData = collectIdsFromNestedData(relationData);
+  const idsFromNestedData = collectIdsFromNestedData(relationApiData);
   const isStructureCorrect = () => idsFromNestedData.length === data.length
-  const isRelationViewEnabled  = layouts[slug].contentType.settings.relation;
-  const isTreeViewPossible = isStructureCorrect() && isRelationViewEnabled;
-  const isRelationErrorVisible = () => isRelationViewEnabled && !isStructureCorrect();
+  const isRelationApiDataCorrect = () => {
+    if(data.length === 0) {
+      return relationApiData.length === 0;
+    }
+    return relationApiData.length !== 0;
+  }
 
+  const isTreeViewPossible = isStructureCorrect() && isRelationViewEnabled;
```

Comment:
maybe reverse operator? 
do this:
```isRelationViewEnabled && isStructureCorrect()```
instead of: 
```isStructureCorrect() && isRelationViewEnabled```

just micro improvement but, boolean field is already calculated, and doesn't need to trigger function if it's false

Reply:
reversed